Add task 1

mekkadarun opened the pull request:
To test out task 1, you may run ./VidDisplay and press 'z'.

anpu9 commented on src/vidDisplay.cpp:
```diff
@@ -34,14 +35,19 @@ class CameraApp {
+    const string WINDOW_THRESHOLD = "Thresholded"; // Name of the thresholded video window
     string editWindow;              // Name of the meme editor window
 
+    bool showThresholded = false; // Flag to indicate whether to show the thresholded video
```
Better approach is moving "showThresholded" to Mode enum

anpu9 commented on src/vidDisplay.cpp:
```diff
@@ -109,6 +115,20 @@ class CameraApp {
+
+            // Apply thresholding to the Value channel
+            threshold(valueChannel, imgs.thresholded);
+            imshow("HSV Thresholded", imgs.thresholded);
```
As metioned if we have a mode::Threshold, we can move the operations into case clause:
e,g
case Mode::THRESHOLD: {
            Mat hsv;
            bgr_to_hsv(imgs.frame, hsv);

            // Extract the Value channel (grayscale)
            Mat valueChannel;
            extractChannel(hsv, valueChannel, 2);  // Value channel is at index 2

            // Apply thresholding to the Value channel
            threshold(valueChannel, imgs.thresholded);
            imshow("HSV Thresholded", imgs.thresholded);
        }

anpu9 replied:
To keep the lines extracted as a function is even better.

anpu9 replied:
hsv mat could be placed into struct imgs to keep clean.

anpu9 commented on src/vidDisplay.cpp:
```diff
@@ -119,6 +139,14 @@ class CameraApp {
         switch (key) {
             case 's': saveImages(); break;
             case 'q': throw runtime_error("User exit");
+            case 'z':
```
case 'z': change it to Thredhold mode
For example: //            case 'g': currentMode = Mode::GREY; break;

anpu9 commented on src/vidDisplay.cpp:
```diff
@@ -128,7 +156,7 @@ class CameraApp {
      */
     CameraApp(int deviceId = 0) {
-        if (!cap.open(deviceId)) {
+        if (!cap.open("/dev/video2")) {                         //Replace with deviceId if needed
```
It is imcompatible for iOS but if's easy to adjust

anpu9 reviewed (changes requested):
Hey Arun, I found we may need to refactor a little bit, do you mind discussing about it?

mekkadarun commented:
I have made all the changes you've mentioned and removed a few unwanted commented lines from vidDIsplay.cpp. Please let me know your thoughts. Task 2 commit is also added.
